service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -314,12 +318,13 @@ protected synchronized void cleanupOperationLog(final long operationLogCleanupDe
       } else {
         if (operationLogCleanupDelayMs > 0) {
           ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
-          scheduledExecutorService.schedule(new OperationLogCleaner(operationLog), operationLogCleanupDelayMs,
+          scheduledExecutorService.schedule(new OperationLogCleaner(this, operationLog), operationLogCleanupDelayMs,
             TimeUnit.MILLISECONDS);
           scheduledExecutorService.shutdown();
         } else {
           log.info("Closing operation log {} without delay", operationLog);
           operationLog.close();
+          OperationLogManager.closeOperation(this);
```

Comment:
nit: can we put `OperationLogManager.closeOperation(this)` into `operationLog.close()`? so we can only take care of SQLOperation's log

Reply:
The reason I did not set inside operationLog.close is that OperationLog is a wrapper of the log file itself and has no  knowledge of the operation.